### ISSUE TITLE
New request: minecraft.wiki

### ISSUE THREAD

**Issue by LMSSonos** [openzim/zim-requests#710]
*Please use the following format for a ZIM creation request (and delete unnecessary information)*

- Website URL: https://minecraft.wiki
- License: CC BY-NC-SA 3.0
- Desired ZIM Title: Minecraft Wiki
- Desired ZIM Description: Wiki about Minecraft
- Desired ZIM Icon –png (URL or attach one): https://minecraft.wiki/images/Wiki.png
- Language (ISO 639-3): eng
- Is this a MediaWiki?: yes

**Comment by Ezkrraa** [comment]
there is a minecraft fandom wiki thread that failed at [https://farm.openzim.org/pipeline/4f2d6add-0f65-4312-937b-6870ce481bad/debug] because of a description of > 80 characters, would love to see this tried again.

**Comment by kelson42** [comment]
@Ezkrraa Minecraft fandom wiki - so in german - last run has been a success. https://farm.openzim.org/recipes/minecraft_fandom_de_all

**Comment by Ezkrraa** [comment]
since the Minecraft community is moving to [minecraft.wiki](minecraft.wiki) I'm expecting this to get updated more over time, and I would love an English version in case my internet goes down :)

**Comment by Markus-Rost** [comment]
Did this request fall through and get missed? It would be great if this could be done. 

Looking at #755 there shouldn't be any upstream issues preventing this for getting done anymore.

**Comment by benoit74** [comment]
It will be done at some point in time, see https://github.com/openzim/zim-requests/issues/755#issuecomment-2720266133

**Comment by surfrock66** [comment]
I've been doing further troubleshooting on this and think I have some progress, but I'm not fully sure I understand the differences in what I'm doing, and might have a separate issue that will be easier to solve.  Looking at some other zimfarm commands, I changed my forced renderer to "RestAPI" and it seems to have mostly completed.  I don't understand the different renderer options and what they do, but this got me to 100%:

```
docker run --volume=/root:/out -ti ghcr.io/openzim/mwoffliner mwoffliner --mwUrl=https://minecraft.wiki/ --adminEmail=[email@domain.com](mailto:email@domain.com) --customZimTitle="Minecraft.wiki" --customZimDescription="The most comprehensive Minecraft wiki on the internet!" --customZimLanguage=eng --mwWikiPath="/w/" --mwActionApiPath="/api.php" --mwRestApiPath="/rest.php" --filenamePrefix=minecraft.wiki --resume --speed=0.1 --forceRender="RestApi"
```

After the logs get to 100% though, I get another error:

```
[log] [2025-04-18T19:15:45.211Z] Downloading module dependencies
[log] [2025-04-18T19:15:45.605Z] Not able to download content for https://minecraft.wiki/w/load.php?debug=true&lang=en&modules=startup&only=scripts&skin=vector&version=&* due to AxiosError: Request failed with status code 404
[error] [2025-04-18T19:15:45.612Z] Failed to run mwoffliner after [15372s]: {
        "message": "Request failed with status code 404",
        "name": "AxiosError",
        "stack": "AxiosError: Request failed with status code 404\n    at settle (file:///tmp/mwoffliner/node_modules/axios/lib/core/settle.js:19:12)\n    at Unzip.handleStreamEnd (file:///tmp/mwoffliner/node_modules/axios/lib/adapt
ers/http.js:589:11)\n    at Unzip.emit (node:events:539:35)\n    at endReadableNT (node:internal/streams/readable:1344:12)\n    at process.processTicksAndRejections (node:internal/process/task_queues:82:21)\n    at Axios.request (fi
le:///tmp/mwoffliner/node_modules/axios/lib/core/Axios.js:45:41)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async getContentCb (file:///tmp/mwoffliner/lib/Downloader.js:459:30)",
        "config": {
```

The logs after this are a very large amount of json, and it's jibberish to me, but I've seen a few people with similar issues that were not able to be resolved.  Is this type of error one I can deal with, or is there a flag to pass to change how dependencies are downloaded?  Additionally, is there more info on what the different renderers do?

**Comment by Markus-Rost** [comment]
> After the logs get to 100% though, I get another error:
> 
> ```
> [log] [2025-04-18T19:15:45.211Z] Downloading module dependencies
> [log] [2025-04-18T19:15:45.605Z] Not able to download content for https://minecraft.wiki/w/load.php?debug=true&lang=en&modules=startup&only=scripts&skin=vector&version=&* due to AxiosError: Request failed with status code 404
> [error] [2025-04-18T19:15:45.612Z] Failed to run mwoffliner after [15372s]: {
>         "message": "Request failed with status code 404",
>         "name": "AxiosError",
>         "stack": "AxiosError: Request failed with status code 404\n    at settle (file:///tmp/mwoffliner/node_modules/axios/lib/core/settle.js:19:12)\n    at Unzip.handleStreamEnd (file:///tmp/mwoffliner/node_modules/axios/lib/adapt
> ers/http.js:589:11)\n    at Unzip.emit (node:events:539:35)\n    at endReadableNT (node:internal/streams/readable:1344:12)\n    at process.processTicksAndRejections (node:internal/process/task_queues:82:21)\n    at Axios.request (fi
> le:///tmp/mwoffliner/node_modules/axios/lib/core/Axios.js:45:41)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async getContentCb (file:///tmp/mwoffliner/lib/Downloader.js:459:30)",
>         "config": {
> ```

Looks like you are trying to access `https://minecraft.wiki/w/load.php` but the script path for the wiki is just `https://minecraft.wiki/load.php`

You need to set the module path using `--mwModulePath="/load.php"`

**Comment by surfrock66** [comment]
That was exactly it, and it's successfully produced the zim which I can browse.  The formatting needs a lot of help, but for an MVP, that did it.  I can share the zim if needed (2.5G) and someone with more experience may know some obvious fixes for the formatting, though maybe that's a matter of the restapi Renderer?

**Comment by jirkacapek123** [comment]
> That was exactly it, and it's successfully produced the zim which I can browse. The formatting needs a lot of help, but for an MVP, that did it. I can share the zim if needed (2.5G) and someone with more experience may know some obvious fixes for the formatting, though maybe that's a matter of the restapi Renderer?

That would be great, please send it.

**Comment by surfrock66** [comment]
~~https://nextcloud.surfrock66.com/s/sgFCxwWZHwNzWtc~~ Here you go, I'll make it available for a few days. (I removed the link for now)

**Comment by alpenglow174** [comment]
Whats the progress on this like?

**Comment by jirkacapek123** [comment]
Unfortunately the progress is almost zero, but I'm trying my best to fix it quickly.

**Comment by benoit74** [comment]
I was curious so I restarted Zimfarm recipe and it looks like it is making good progress: https://farm.openzim.org/pipeline/9c9e9191-1339-4b70-aba7-b409c2f1e046

Might consider to do all minecraft wikis soon if confirmed.

**Comment by benoit74** [comment]
There are still significant visual issues but this seems to look a lot like an issue impacting wikipedia.org website which is obviously the highest priority, so it will probably be solved very quickly.

**Comment by surfrock66** [comment]
If you are ok with having it be a little visually messy, using the restAPI renderer produces a usable wiki which has all the content, and here's my command to generate it; update your email address of course:

```
date=$(date +%Y-%m-%d_%H-%M-%S); wikiname="minecraft.wiki"; echo $date; echo $date > mwoffliner.$wikiname.$date.log; docker run --volume=/etc/kiwix/data:/out -ti ghcr.io/openzim/mwoffliner mwoffliner --mwUrl=https://minecraft.wiki --adminEmail=___@___.com --customZimTitle="Minecraft.wiki" --customZimDescription="The most comprehensive Minecraft wiki on the internet!" --customZimLanguage=eng --mwWikiPath="/w/" --mwActionApiPath="/api.php" --mwRestApiPath="/rest.php" --mwModulePath="/load.php" --filenamePrefix=minecraft.wiki --resume --speed=0.1 --forceRender="RestApi" 2>&1 > mwoffliner.$wikiname.$date.log
```

**Comment by surfrock66** [comment]
> I was curious so I restarted Zimfarm recipe and it looks like it is making good progress: https://farm.openzim.org/pipeline/9c9e9191-1339-4b70-aba7-b409c2f1e046
> 
> Might consider to do all minecraft wikis soon if confirmed.

When I run the help of mwoffliner, I see this line:
```
   --forceRender               Force the usage of a specific API end-point/render, automatically chosen otherwise. Accepted values: [ VisualEditor, WikimediaDesktop. WikimediaMobile, RestApi ]
```
In the zimfarm recipe you're using, I see ActionParse as the renderer, but I'm not seeing anything about that, what is that renderer and what is the case to use it?

**Comment by Markus-Rost** [comment]
> I was curious so I restarted Zimfarm recipe and it looks like it is making good progress: https://farm.openzim.org/pipeline/9c9e9191-1339-4b70-aba7-b409c2f1e046
> 
> Might consider to do all minecraft wikis soon if confirmed.

I noticed some issues with the config of the recipe. 
1. The icon is wrongly using a Wikia/Fandom image instead of https://minecraft.wiki/images/Wiki.png
2. When browsing the file, the footer claims the articles were issued by "Minecraft" instead of "Minecraft Wiki".
3. The license is incorrectly set to CC BY-SA 4.0 but should be CC BY-NC-SA 3.0 instead.

**Comment by benoit74** [comment]
> If you are ok with having it be a little visually messy, using the restAPI renderer produces a usable wiki which has all the content, and here's my command to generate it; update your email address of course:

I think the one which was just produced in Zimfarm is complete and "usable" as well. Or did you spotted some issues (besides visual problems).

> In the zimfarm recipe you're using, I see ActionParse as the renderer, but I'm not seeing anything about that, what is that renderer and what is the case to use it?

This renderer has been introduced in 1.15.0. You will find little more info about it at https://github.com/openzim/mwoffliner/wiki/API-end%E2%80%90points. It is ought to become the future default renderer in 1.16 (or a bit later) because it is deemed to now be available in all mediawikis since quite a long time. It is also the only renderer where a real and clean support of the Mediawiki skins has been implemented (before that skin mgmt was never correctly understood and mostly a mess, and we've not backported support to other renderers, since it might be a waste should we decide to not support them anymore). For now only vector (legacy and 2022) skins are implemented, but it is only a small effort to support more skins. 

> I noticed some issues with the config of the recipe.

Thanks a lot ! I've removed the custom icon (I suspect we can use default one coming from website), I've opened https://github.com/openzim/mwoffliner/issues/2344 and https://github.com/openzim/mwoffliner/issues/2345 about footer content.

**Comment by surfrock66** [comment]
Good to know that is the new default, ActionParse and RestAPI renderer look nearly the same, except ActionParse defaults to a wider display which is nice.  I assume the CSS thing is visualeditor only, or is that some quirk of this site?

**Comment by alpenglow174** [comment]
I noticed the crafting recipes are broken. Is this a bug or just a limitation?

<img src="https://github.com/user-attachments/assets/361ac9ef-f089-4ef4-8fd8-402255b44260" height="300">
<img src="https://github.com/user-attachments/assets/95a3b1dc-2170-47ad-8d39-42b9f1640154" height="300">

**Comment by Markus-Rost** [comment]
> I noticed the crafting recipes are broken. Is this a bug or just a limitation?

This is caused by CSS gadgets not being loaded, see openzim/mwoffliner#2212

**Comment by kelson42** [comment]
Better in latest scrape? https://farm.openzim.org/recipes/minecraft_wiki_en_all

**Comment by Markus-Rost** [comment]
`customZimFavicon` is still wrong in the config.

**Comment by alpenglow174** [comment]
https://github.com/user-attachments/assets/9aabc6c5-17a7-4fe4-ba2c-b956368370fc

Looks like some stuff is broken?

**Comment by violine1101** [comment]
Do other pages work properly or is it just the main page? It seems like all links in general are disabled.

Not entirely sure what's going on with the faulty rendering. It seems like some CSS rules don't apply properly and/or your browser (or whatever you're using to view the wiki) doesn't properly support some CSS features we're using.

**Comment by alpenglow174** [comment]
I tried it using pwa.kiwix.org on both chrome and firefox. Seems to behave to same. I checked other pages and the links are disabled there too. Also weird formatting with the recipes.

<img width="856" height="393" alt="Image" src="https://github.com/user-attachments/assets/bceffd17-5d7e-4fa1-8943-a18e6e64c816" />

**Comment by Markus-Rost** [comment]
The weird formatting with recipes looks like it's the empty inventory slots. @benoit74 are we removing empty elements?

No idea regarding all the missing links.

**Comment by Markus-Rost** [comment]
> Better in latest scrape? https://farm.openzim.org/recipes/minecraft_wiki_en_all

Looks like the ZIM is using vector-2022 but the wiki only supports vector-legacy. 

Needs another scrape once https://github.com/openzim/mwoffliner/issues/2358 is done.

**Comment by benoit74** [comment]
Will have to debug, looks like there are still some rough corner cases to fix ^^

**Comment by benoit74** [comment]
There is a new version available which looks pretty decent, all links are working well. Some styling CSS is still misses and causes some weird things here and there, but from my PoV this is acceptable for a first version.

@kelson42 @Markus-Rost and others, WDYT? Do you achieve to spot something which is too significantly broken (besides homepage overlaps)

**Comment by benoit74** [comment]
Oh no, I mostly forgot about https://github.com/openzim/mwoffliner/issues/2415 ; we must fix this, it is just too important for crafting recipes.

**Comment by Markus-Rost** [comment]
Looking at the latest scrape https://farm.openzim.org/pipeline/14eb083e-c665-4e84-bd9d-2007563468a4 I see most of the issues fixed. However I also noticed some new issues:

- Selflinks are missing (openzim/mwoffliner#2454)
- All sound files are missing (openzim/mwoffliner#2455)
- All SVG files fail to load when viewing an article. The link to the file is correct, I can view and download the svg file directly just fine. 
  - @kelson42 I'm unsure where that issue should be reported to. I got it with both kiwix-js and kiwix-android but it seems unrelated to mwoffliner.
  - https://minecraft.wiki/ <br><img width="422" height="239" alt="Image" src="https://github.com/user-attachments/assets/de09726d-232f-4144-ab5b-752cd3c1a947" />
  - https://minecraft.wiki/w/Logo#Variations <br><img width="687" height="896" alt="Image" src="https://github.com/user-attachments/assets/a8c5aa08-b796-4b83-9580-1b092c058a91" />

**Comment by Markus-Rost** [comment]
The recipe should probably be updated to remove the `novid` format so that sound files will be included.
https://farm.openzim.org/recipes/minecraft.wiki_en_all

**Comment by kelson42** [comment]
> @kelson42 I'm unsure where that issue should be reported to. I got it with both kiwix-js and kiwix-android but it seems unrelated to mwoffliner.

Thanks a lot for testing! Should be reported once per reader.

**Comment by Markus-Rost** [comment]
> > [@kelson42](https://github.com/kelson42) I'm unsure where that issue should be reported to. I got it with both kiwix-js and kiwix-android but it seems unrelated to mwoffliner.
> 
> Thanks a lot for testing! Should be reported once per reader.

I have created kiwix/kiwix-js#1357 and kiwix/kiwix-android#4366

**Comment by Popolechien** [comment]
Other than the above, shouldn't this issue be considered as a duplicate of #544 ?

**Comment by Markus-Rost** [comment]
@benoit74 I finally got to reworking our JS modules to make them safe to use in mwoffliner. Please add the following `addModules` to the recipe for use once mwoffliner 2.0 has released:
`site,ext.gadget.layeredBlueprint,ext.gadget.refTooltip,oojs,oojs-ui,ext.gadget.sound,ext.gadget.simplecalc-loader,ext.gadget.simplecalc,@wikimedia/codex` 

Creating a new ZIM file with mwoffliner 2.0 at that point would be appreciated as well.

**Comment by benoit74** [comment]
@Markus-Rost thank you very much, very appreciated.

We are in the final stage of mwoffliner 2.0 and are testing it on multiple Zimfarm recipes. Sorry for not having thought about including this one ... Now done, with all the modules you've whitelisted for us. Zimfarm task is running, let's see.

**Comment by benoit74** [comment]
Just fixed the recipe to request "full" rather than "maxi", as stated above we need this to have sounds
